content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+There are two styles of pagination that are commonly used, a "page-by-page" style---where you show only one page of results at a time, starting at some offset (which the user can control), and "infinite-scroll" style, where you show and increasing number of pages of items, as the user moves through the list (this is the typical "feed" style user interface).
+
+Let's consider a publication/subscription technique for the second (the first technique is quite difficult to do correctly in Meteor as you can't tell at which offset to start on the client-side which only has a subset of the true data).
+
+XXX: Should we deal with the page-to-page pagination scenario better?
+
+In an infinite scroll publication, we simply need to add a new argument to our publication controlling how many items to load. Suppose we wanted to paginate the todo items in our Todos example app:
+
+```js
+const MAX_TODOS = 1000;
+
+Meteor.publish('list/todos', function(listId, limit) {
+  check(listId, String);
+  check(limit, Number);
+
+  const options = {
+    sort: {createdAt: -1},
+    limit: Math.min(limit, MAX_TODOS)
+  };
+
+  ...
+});
+```
+
+It's important that we set a `sort` parameter on our query (after, which first `limit` todos do we want?), and that we set an absolute maximum on the number of items a user can request (at least in the case where lists can grow without bound).
+
+Then on the client side, we'd some kind of reactive state variable to control how many items to request:
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.set('listId', FlowRouter.getParam('_id'));
+    this.subscribe('list/todos', this.state.get('listId'), this.state.get('requestedTodos'));
+  });
+});
+```
+
+We'd increment that `requestedTodos` variable when the user clicks "load more" (or perhaps just when they scroll to the bottom of the page).
+
+Once piece of information that's very useful to know when paginating data is the *total number of items* that you could see. The `[tmeasday:publish-counts](https://atmospherejs.com/tmeasday/publish-counts)` can be useful to publish this. We could add a `/list/todoCount` publication like so
+
+```js
+Meteor.publish('list/todoCount', function(listId) {
+  check(listId, String);
+  
+  Counts.publish(this, `list/todoCount${listId}`, Todos.find({listId}));
+});
+```
+
+Then on the client, after subscribing to that publication, we can access the count with `Counts.get(``list/todoCount${listId}`)`.
+
+## Other (non-published) data sets: Stores
+
+In Meteor, persistent or shared data comes over the wire on publications. However, there are some types of data which doesn't need to be persistent or shared between users. For instance, the "logged-in-ness" of the current user, or the route they are currently viewing. 
```

Comment:
Oh I actually wanted the two definitions here because I wanted to emphasise there are two reasons to use a pub -- either b/c the data needs to persist, or is shared.